### PR TITLE
feat(processors): Add metrics tag cardinality limit processor

### PR DESCRIPTION
**Pull request by darinspivey**
This adds a TF resource for Pipeline's processor,
`mezmo-tag-cardinality-limit`.

Ref: LOG-18369

### CODE REVIEW

File: internal/provider/models/processors/test/metrics_tag_cardinality_limit_test.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+					StateHasExpectedValues("mezmo_metrics_tag_cardinality_limit_processor.my_processor", map[string]any{
+						"pipeline_id":   "#mezmo_pipeline.test_parent.id",
+						"title":         "title",
+						"description":   "desc",
+						"generation_id": "0",
+						"inputs.#":      "0",
+						"mode":          "exact",
+						"action":        "drop_event",
+						"value_limit":   "50",
+						"%":             "11",
```
Comment (biblicalph) [inline internal/provider/models/processors/test/metrics_tag_cardinality_limit_test.go:144]: What does "%" here refer to?

**Review by biblicalph — COMMENTED** [review]
Just the one question.
Looks good

File: internal/provider/models/processors/test/metrics_tag_cardinality_limit_test.go (hunk 1 above)
Comment (darinspivey) [reply internal/provider/models/processors/test/metrics_tag_cardinality_limit_test.go:144]: it's the total number of key/vals.  Adding this to the assertion makes sure the server hasn't added properties we don't have covered.

**Review by jorgebay — APPROVED** [review]
Nothing to add, looks great 👍

**Review by biblicalph — APPROVED** [review]
Thanks for clarifying looks good

**Pull request merged**